### PR TITLE
Upgrade all dependencies related with Babel

### PR DESCRIPTION
**Pull request by jhonny218**
## Description
In this PR all dependencies related with Babel will be upgraded.

All upgrades are minors except `babel-jest` and `babel-plugin-dynamic-import-node` with a major upgrade.

## Note:
- `babel-jest` went from `23.4.2` to `27.0.6`, 4 versions up. Based on their [releases](https://github.com/facebook/jest/releases), no breaking changes were introduced.
- `babel-plugin-dynamic-import-node` went from `1.2.0` to `2.3.3`. Based on their [releases](https://github.com/airbnb/babel-plugin-dynamic-import-node/releases), no breaking changes were introduced.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#28952

## Testing done
Locally

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by alexandec** [comment]
Probably not an issue since these aren't major upgrades, but I recommend testing with IE 11 just to make sure there aren't any issues with polyfills, transpilation, etc. Our automated tests won't check anything in IE 11.

**Review by oseasmoran73 — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -32,7 +32,7 @@
     "lint:js:changed:fix": "LIST=`git diff-index --diff-filter=d --name-only HEAD | grep \"\\.js[x]\\{0,1\\}$\"`; if [ \"$LIST\" ]; then eslint --fix --quiet $LIST; fi",
     "lint:js:untracked:fix": "LIST=`git ls-files --others --exclude-standard | grep \"\\.js[x]\\{0,1\\}$\"`; if [ \"$LIST\" ]; then eslint --fix --quiet $LIST; fi",
     "lint:sass": "stylelint src/**/*.scss",
-    "lint:sass:changed": "LIST=`git diff-index --diff-filter=d --name-only HEAD -- \"*.scss\"`; if [ \"$LIST\" ]; then stylelint $LIST; fi", 
+    "lint:sass:changed": "LIST=`git diff-index --diff-filter=d --name-only HEAD -- \"*.scss\"`; if [ \"$LIST\" ]; then stylelint $LIST; fi",
```
Comment (erikphansen) [inline package.json:35]: 🚀  😆

**Review by erikphansen — APPROVED** [review]
A bunch of version bumps without needing to change any code? 👍

**Review by Rleahy22 — APPROVED** [review]
Needs a rebase but LGTM!

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Review by LeKeve — APPROVED** [review]

**Review by micahchiang — APPROVED** [review]
Tested my team's apps in the review instance. LGTM 👍

**Comment by adamwhitlock1** [comment]
LGTM 💯

**Review by mpelzsherman — APPROVED** [review]
Tested facilities in RI - LGTM

**Review by adamwhitlock1 — APPROVED** [review]

**Review by Mottie — APPROVED** [review]

**Review by bosawt — APPROVED** [review]
Looks good! I confirmed an SSOe auth still works as expected, didn't seem to break any part of the login flow

**Review by bkjohnson — APPROVED** [review]
These dependency upgrades don't seem like they would break forms/design system things. I was curious about the two major version upgrades, but based on the name it seems like they would just affect tests and other dev scripts, so if something broke I'd expect CI to let us know.

**Review by CBonade — APPROVED** [review]
LGTM!

**Review by dillo — APPROVED** [review]

**Review by stephenBarrs — APPROVED** [review]

**Review by jgacek — APPROVED** [review]

**Review by vbahinwillit — COMMENTED** [review]
LGTM!

**Pull request merged**